src/simulation/m_global_parameters.fpp

```diff
@@ -1142,19 +1176,24 @@ contains
         xiend = xi_idx%end
         chemxb = species_idx%beg
         chemxe = species_idx%end
+        Bxb = B_idx%beg
+        Bxe = B_idx%end
 
-        !$acc update device(momxb, momxe, advxb, advxe, contxb, contxe, bubxb, bubxe, intxb, intxe, sys_size, buff_size, E_idx, alf_idx, n_idx, adv_n, adap_dt, pi_fac, strxb, strxe, chemxb, chemxe)
+        !$acc update device(momxb, momxe, advxb, advxe, contxb, contxe, bubxb, bubxe, intxb, intxe, sys_size, buff_size, E_idx, alf_idx, n_idx, adv_n, adap_dt, pi_fac, strxb, strxe, chemxb, chemxe, Bxb, Bxe)
```

Comment:
I'm beginning to wonder if the new compilers really need the change from mom_idx to momxb, B_idx%beg to Bxb, etc. can you try taking this out? if it works, can you perform the more intrusive change of removing it in more places (checking that it isn't breaking things on nvidia or Frontier/amd.